shared/django_apps/codecov_metrics/service/codecov_metrics.py

```diff
@@ -24,7 +24,7 @@ def create_user_onboarding_metric(org_id: int, event: str, payload: Dict[str, An
             return
 
         if not UserOnboardingLifeCycleMetrics.objects.filter(
-            org_id=org_id, event=event
+            org_id=org_id, event=event, additional_data=payload
```

Comment:
For this entire part I think you can just use `get_or_create`

```
UserOnboardingLifeCycleMetrics.objects.get_or_create(
    org_id=org_id,
    event=event,
    additional_data=payload,
)
```

Reply:
Oh cool, ty